packages/ifpack2/src/Ifpack2_BlockTriDiContainer_decl.hpp

```diff
@@ -109,21 +109,18 @@ namespace Ifpack2 {
       //! complex plane. N.B. that this constant modifies the matrix in the linear
       //! equation, not simply the diagonal preconditioner.
       magnitude_type addRadiallyToDiagonal = Kokkos::ArithTraits<magnitude_type>::zero();
-      // ComputeParameters() 
-      //   : addRadiallyToDiagonal(Kokkos::ArithTraits<magnitude_type>::zero()) 
-      // {}
     };
 
     //! Input arguments to <tt>applyInverseJacobi</tt>
     struct ApplyParameters {
       //! Set this to true if <tt>Y</tt> is meant to be treated as 0 on
       //! entry. Defaults to false.
-      bool zeroStartingSolution;
+      bool zeroStartingSolution = false;
       //! Damping factor. Defaults to 1.
-      scalar_type dampingFactor;
+      scalar_type dampingFactor = Kokkos::ArithTraits<magnitude_type>::one();
```

Comment:
`scalar_type` or `magnitude_type`?

Reply:
I do not template scalar_type and magnitude_type. Both types are double. The reason I do not template them is because the derived types from MatrixType parameter can be a UQ type (UQ type tolerance value does not make sense). Also, UQ type instantiation is not supported in this code (it uses SIMD packed format and it is not combined with UQ type) but it has to be compiled for Stokhos ETI. Instead of making the code more complicated, I choose to use double for parameters i.e., radial to diagonals, damping factor, and tolerance, which can be worked for most use cases (in fact we only instantiate for double and that is the only use case). Later when I improve this code again, I can fix this typo but it does not cause a problem in this PR.

Reply:
@kyungjoo-kim wrote:
> I do not template scalar_type and magnitude_type. Both types are double.

Sierra builds Ifpack2 for `scalar_type=complex<double>` too.  It's true that the assignment should work in that case, but it's valuable to get straight whether `dampingFactor` is a scalar or a magnitude.